src/Sylius/Component/Promotion/Checker/CouponsEligibilityChecker.php

```diff
@@ -46,8 +46,6 @@ public function isEligible(PromotionSubjectInterface $subject, PromotionInterfac
         }
 
         if (!$this->isCouponEligible($promotion, $subject)) {
-            $this->eventDispatcher->dispatch(SyliusPromotionEvents::COUPON_NOT_ELIGIBLE, new GenericEvent($promotion));
```

Comment:
This isn't anything to do with rules though. This is just checking to ensure the code you entered is the same as the one attached to the promotion. 

We definitely need a change to the Rules section though to add what you are proposing.